nixos/modules/system/boot/loader/raspberrypi/uboot-builder.sh

```diff
@@ -21,18 +22,37 @@ copyForced() {
 
 # Add the firmware files
 fwdir=@firmware@/share/raspberrypi/boot/
+copyForced $fwdir/overlays  $target/overlays
 copyForced $fwdir/bootcode.bin  $target/bootcode.bin
 copyForced $fwdir/fixup.dat     $target/fixup.dat
+copyForced $fwdir/fixup4.dat    $target/fixup4.dat
+copyForced $fwdir/fixup4cd.dat  $target/fixup4cd.dat
+copyForced $fwdir/fixup4db.dat  $target/fixup4db.dat
+copyForced $fwdir/fixup4x.dat   $target/fixup4x.dat
 copyForced $fwdir/fixup_cd.dat  $target/fixup_cd.dat
 copyForced $fwdir/fixup_db.dat  $target/fixup_db.dat
 copyForced $fwdir/fixup_x.dat   $target/fixup_x.dat
 copyForced $fwdir/start.elf     $target/start.elf
+copyForced $fwdir/start4.elf    $target/start4.elf
+copyForced $fwdir/start4cd.elf  $target/start4cd.elf
+copyForced $fwdir/start4db.elf  $target/start4db.elf
+copyForced $fwdir/start4x.elf   $target/start4x.elf
 copyForced $fwdir/start_cd.elf  $target/start_cd.elf
 copyForced $fwdir/start_db.elf  $target/start_db.elf
 copyForced $fwdir/start_x.elf   $target/start_x.elf
 
+# Add the config.txt
+copyForced @configTxt@ $target/config.txt
+
 # Add the uboot file
 copyForced @uboot@/u-boot.bin $target/u-boot-rpi.bin
 
-# Add the config.txt
-copyForced @configTxt@ $target/config.txt
+# TODO: should we be copying over the DTBs for the other pis? how are we getting away with that
+# my understanding is that these are needed for u-boot to even load, so?
```

Comment:
My understanding here is that the pi3 family FDTs setup by the firmware have everything u-boot needs to continue booting.

I'm not sure what's missing from the firmware setup for the pi4 family :/.

Reply:
Good enough for me, comment removed/replaced